### PR TITLE
(#198) unit tests for transports

### PR DESCRIPTION
**Pull request by joeldart**
(#198) bootstrapping unit tests for transports

### CODE REVIEW

**Comment by joeldart** [comment]
@abillingsley I have at least two tests in here that are just failing as not implemented, but I think this can be used to build out more deterministic control flow and track down more of these reconnecting edge cases

Review/thoughts?

**Comment by joeldart** [comment]
edit: squashed temp commit

**Comment by abillingsley** [comment]
@joeldart looks good to me

File: SignalR.Client/SRConnectionInterface.h [hunk 1]
```diff
@@ -49,6 +49,12 @@
 @property (strong, nonatomic, readwrite) NSMutableDictionary *headers;
 
 ///-------------------------------
+/// @name Internal state
+///-------------------------------
+
+- (NSString *)onSending;//TODO: this just encapsulates connectionData. can we pull this into a getUrl like js client does?
```
Comment (joeldart) [inline SignalR.Client/SRConnectionInterface.h]: We can remove this from the interface bc we were able to get control flow working nicely inside stop

**Comment by joeldart** [comment]
@abillingsley branch up to date for investigating the ARC issues we talked about
